scripts/crc-deploy.sh

```diff
@@ -0,0 +1,192 @@
+# it creates a helm values files and other manifest patch files applied by
+# scripts/generate_crds_manifests.sh
+function make_manifest_patches {
+  OLM_IMG=${1?Error:\ olm image undefined}
+  OPM_IMG=${2?Error:\ opm image undefined}
+  SED_OPTS=(-e 's#OLM_IMAGE#'"${OLM_IMG}"'#g' -e 's#OPM_IMAGE#'"${OPM_IMG}"'#g')
+
+  # helm values file
+  VALUES_TEMPLATE=$(mktemp)
+  cat << EOF > "${VALUES_TEMPLATE}"
+olm:
+  image:
+    ref: OLM_IMAGE
+catalog:
+  commandArgs: --configmapServerImage=OPM_IMAGE
+  opmImageArgs: --opmImage=OPM_IMAGE
+  image:
+    ref: OLM_IMAGE
+package:
+  image:
+    ref: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${VALUES_TEMPLATE}" > "${CRC_E2E_VALUES}"
+
+  # psm operator patch
+  PSM_OPERATOR_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${PSM_OPERATOR_PATCH_TEMPLATE}"
+- command: update
+  path: spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+- command: update
+  path: spec.template.spec.containers[0].env[1].value
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${PSM_OPERATOR_PATCH_TEMPLATE}" > scripts/psm-operator-deployment.crc.e2e.patch.yaml
+
+  # collect-profiles patch
+  COLLECT_PROFILE_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${COLLECT_PROFILE_PATCH_TEMPLATE}"
+- command: update
+  path: spec.jobTemplate.spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${COLLECT_PROFILE_PATCH_TEMPLATE}" > scripts/collect-profiles.crc.e2e.patch.yaml
+}
+
+YQ="go run ./vendor/github.com/mikefarah/yq/v3/"
+export CRC_E2E_VALUES="values-crc-e2e.yaml"
+
+# Set kubeconfig to CRC if necessary
+export KUBECONFIG=${KUBECONFIG:-${HOME}/.crc/machines/crc/kubeconfig}
+KUBE_ADMIN_LOGIN=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.username)
+KUBE_ADMIN_PASSWORD=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.password)
+
+# login to crc
+echo "Logging in as kubeadmin"
+oc login -u "${KUBE_ADMIN_LOGIN}" -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
+
+# Scale down CVO to stop changes from returning to stock configuration
+echo "Scaling down CVO"
+oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+
+echo "Allowing cluster wide access to openshift repository"
+oc policy add-role-to-group system:image-puller system:serviceaccounts --namespace=openshift
+
+# push olm images to crc global repository
+push_images olm:test opm:test
+
+SKIP_MANIFESTS=${SKIP_MANIFESTS:-0}
+if [ "${SKIP_MANIFESTS}" = 0 ]; then
+  # Create values and patches files
+  # Get images with the specific shas
+  OLM_IMG="$(oc get istag/olm:latest -o json | jq -r .image.dockerImageReference)"
+  OPM_IMG="$(oc get istag/opm:latest -o json | jq -r .image.dockerImageReference)"
+
+  make_manifest_patches "${OLM_IMG}" "${OPM_IMG}"
+
+  # Build e2e manifests
+  echo "Generating manifests"
+  # CRC_E2E_VALUES is already set in this script and exported
+  ./scripts/generate_crds_manifests.sh
+
+  # Apply patches
```

Comment:
I think we should apply the OLM manifests first and then apply the psm and profiler deployments.

Reply:
Interestingly enough, it looks like do the exact opposite in the context of CVO run levels looking at our root manifests directory: https://github.com/openshift/operator-framework-olm/tree/master/manifests.

Reply:
00 is higher priority right?

Reply:
@dinhxuanvu that's right. I'm updating the script to pipe to sort. Then it should apply everything in priority order. Good catch! The comment here was stale. What I was doing before was deleting them in reverse priority order then applying them in priority order. But then I discovered replace.